api/odl.py

```diff
@@ -423,8 +434,16 @@ def _fulfill(self, loan):
 
         expires = doc.get("potential_rights", {}).get("end")
         expires = datetime.datetime.strptime(expires, self.TIME_FORMAT)
-        content_link = doc.get("links", {}).get("license", {}).get("href")
-        content_type = doc.get("links", {}).get("license", {}).get("type")
+
+        links = doc.get("links", [])
+        content_link = None
+        content_type = None
+        for link in links:
+            if link.get("rel") == "license":
+                content_link = link.get("href")
+                content_type = link.get("type")
+                break
+
```

Comment:
This seems like a great case for a helper method similar to `OPDSImporter.extract_link`. But the only reason I'd say this is useful, as opposed to a good way to save code, is if these `href`s might be relative links that need to be made into absolute URLs.

Are relative links allowed in ODL? The closest thing I can find to an answer is that Readium Web Publication Manifest says that in a standalone manifest, "all resources in readingOrder, resources and links must be referenced relatively to the manifest".

Reply:
This part is actually the LCP License Status Document. I'm not sure if it allows relative links.

Reply:
Sorry, this is actually ODL - but other places in the file that do this same thing are for the LCP License Status Document.

Reply:
Maybe this is a task for later, when we want a library that can turn all of these similar formats -- OPDS 2, ODL, LCP LSD -- into a Python data model.